java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -148,6 +150,7 @@ class EntityHandleManager(
             HandleMode.Read -> object : ReadSingletonHandle<T> by singletonHandle {}
             HandleMode.Write -> object : WriteSingletonHandle<T> by singletonHandle {}
             HandleMode.ReadWrite -> object : ReadWriteSingletonHandle<T> by singletonHandle {}
+            else -> throw Error("Singleton Handles do not support mode ${spec.mode}")
```

Comment:
Don't throw base exception classes like Error, Exception or RuntimeException. Instead, try to find an applicable class to the situation.. in this situation, `IllegalArgumentException` is probably the best choice.